### PR TITLE
Feat/stack-cards

### LINKED ISSUE #5385: Stacking / grouping items from same collection in explorer / bundles / spammy collection
In explorer when we are showing items, it's pretty useless for user to see **many same items** from one collection and it acts as visual spam.  "antispam card" 

I propose these filters in particular order
- using same IPFS hash - that classify definitely will be same visual item
- same price - indicates from one serial mint - if someone sets same price from minting, whatever.
- only one transfer (can still be some kind of airdrop from whitelist/allowlist)
- any other filters?

📟 @vikiival I guess we would need resolver for this kind of inspection on explorer to get reply faster from squid?

## Design
- we are going to replace multiple duplicated content with one card
image
- updated explore v0.9 figma: https://www.figma.com/file/OwAEyE0QDPn3tHXZbyot1o/Explore-v0.9?node-id=0%3A1&t=ezFOQUs4vhM0Txeu-1
image

### Basics 
- since we don't have a listings tab, this card will be only visible in explore/items
- card will show the name of the duplicate nft - when nft would be the same (img, description...) except name (Name #.23 and name #.45) is it possible to use this card as well - but without the number at the end? 
- how many times the nft is duplicated 
- floor price of all the same nfts
- collection name, the imgs are from 
- visit button 
- there is also small card:
  - used only on smaller previews 
  - absence of floor price, otherwise working similarly

### Functionality
- the nft card will work the same as other nft card
- stroke to the most contrast color in the mode + opacity overlay
- if you click the nft img, it will redirect you to the collection
- same with collection name and view 
- basically it's a shortcut to collection 

I would propose start doing this globally or at least on rmrk and rmrk2 as there is **massive amount of spam items** and we want to highlight good collections 

image

---
can test with 
https://beta.kodadot.xyz/rmrk2/u/DkjzQrt7U5WN66STqXmXeQYZkcd2mZuUqzncjXwDCck2wUQ?tab=collection

---
Lmk if you have any questions 🚀

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

@exezbcz please review

## Context

- [x] Closes #5385

**works only on ahk**

need to implement `TokenEntity` on Rubik indexer https://github.com/kodadot/rubick/issues/273

#### Did your issue had any of the "$" label on it?
$

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshots
image
image
image
image

### <samp>🤖 Generated by Copilot at 3ee2b79</samp>

This pull request implements the UI and logic for displaying and interacting with stacked NFTs on the explore items page. It adds and updates components, composables, types, and queries to support the new feature. It also improves the code style, readability, and type safety of some existing files. The main files affected are `ItemsGridImage.vue`, `useItemsGrid.ts`, `useNft.ts`, `NeoNftCard.vue`, and `nftListWithSearch.graphql`.

### <samp>🤖 Generated by Copilot at 3ee2b79</samp>

> _We are the masters of the `NFTMediaInfo`_
> _We extract, reuse, and stack the NFTs_
> _We query, fetch, and display the token entities_
> _We refactor, improve, and shadow the `NeoNftCard`_

### CODE REVIEW

**Comment by daiagi** [comment]
Build failing...

**Comment by preschian** [comment]
> Build failing...

`fragment nft` conflicts with this one https://github.com/kodadot/nft-gallery/blob/main/queries/fragments/nft.graphql

File: queries/subsquid/ahk/nftListWithSearch.graphql
Comment (preschian) [inline queries/subsquid/ahk/nftListWithSearch.graphql:87]: no need to include `__typename` I guess

Comment (daiagi) [reply queries/subsquid/ahk/nftListWithSearch.graphql]: yep. done

**Comment by exezbcz** [comment]
nice job! 
image
- the animation is somewhat weird, see the stroke there. 

why these imgs are not stacked? The filter is not applied or? 
image
https://deploy-preview-7269--koda-canary.netlify.app/ahp/explore/items?listed=true

**Comment by daiagi** [comment]
>the animation is somewhat weird, see the stroke there.

yes, due to opacity of card  on hover

>why these imgs are not stacked? The filter is not applied or?

currently works only on ahk but ahp used same indexer ill check if it can be applied there too

**Comment by daiagi** [comment]
oh.
unique nfts are not visible with new query
https://canary.kodadot.xyz/ahp/gallery/u-8-262

probably fix on indexer side

**Comment by prury** [comment]
Overall looks really nice! well done! 
Some stacked card issue: 
image

File: components/items/ItemsGrid/useItemsGrid.ts
Comment (Jarsen136) [inline components/items/ItemsGrid/useItemsGrid.ts:74]: Why is it only used on the explore page?

File: queries/subsquid/ahk/nftListWithSearch.graphql
Comment (Jarsen136) [inline queries/subsquid/ahk/nftListWithSearch.graphql:41]: Could you please use something fragments like this? This can avoid enumerating all values.

```
#import "../../fragments/nft.graphql"
#import "../../fragments/nftDetails.graphql"
```

File: components/items/ItemsGrid/useItemsGrid.ts
Comment (daiagi) [reply components/items/ItemsGrid/useItemsGrid.ts:74]: See the isuue.
Exec asked for only explore page
If it were to be in collection page as well there would be no way to see individual items

But mostly coz @exezbcz said so

Comment (exezbcz) [reply components/items/ItemsGrid/useItemsGrid.ts:74]: Correct, no listings tab - you need to see the individual items somewhere

**Comment by daiagi** [comment]
> the animation is somewhat weird, see the stroke there.

@exezbcz  resolved

>Some stacked card issue:

@prury  resolved

**Comment by exezbcz** [comment]
> > the animation is somewhat weird, see the stroke there.
> 
> yes, due to opacity of card on hover image
> 
> I think we will need design solution for this @exezbcz
> 
> > why these imgs are not stacked? The filter is not applied or?
> 
> currently works only on ahk. indexer work need to be done on rubick (ksm/rmrk) and speck (ahp) to enable the feature there as well

hmmm, thought that its done through an overlay, adding solid with some % opacity does not work?

**Comment by exezbcz** [comment]
> > the animation is somewhat weird, see the stroke there.
> 
> @exezbcz resolved
> 
> > Some stacked card issue:
> 
> @prury resolved

ah, see that you solved it

**Comment by exezbcz** [comment]
how hard would it be to enable it on profile as well? 

>with listing, I would enable this as well.
On profile, its showing items you own, imagine you have like 300 of them (no matter whether you are the owner or creator + owner), you want to list them.

>When clicking this stacked img, that represents all the items you own and are the same, all the item that don't have price will be added into mass list popup.
then the flow is the same
wdyt?

**Comment by daiagi** [comment]
> how hard would it be to enable it on profile as well?

Not hard

**Comment by yangwao** [comment]
Can we make this one for the next release, like 2-3 days?

**Comment by daiagi** [comment]
@yangwao yes definitely

File: queries/subsquid/ahk/nftListWithSearch.graphql
Comment (daiagi) [reply queries/subsquid/ahk/nftListWithSearch.graphql]: done

**Comment by daiagi** [comment]
>how hard would it be to enable it on profile as well?
done

**Review by Jarsen136 — APPROVED** [review]

File: composables/useNft.ts
Comment (preschian) [inline composables/useNft.ts:61]: can we force it by adding `as NFTWithMetadata` or maybe something else in here?

ts warning on this side 
image

File: libs/ui/src/components/NeoNftCard/NFTMediaInfo.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NFTMediaInfo.vue:9]: fix ts warning? since `nft?` is optional props

image

File: libs/ui/src/components/NeoNftCard/NFTMediaInfo.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NFTMediaInfo.vue:71]: how about this? no need to check `props.variant` first, because already had 'primary' as default value

```suggestion
const isMinimal = computed(() => props.variant.includes('minimal'))
```

File: libs/ui/src/components/NeoNftCard/NFTMediaInfoStacked.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NFTMediaInfoStacked.vue:56]: use `NFTStack` instead?

```suggestion
    nft: NFTStack
```

File: libs/ui/src/components/NeoNftCard/NFTMediaInfoStacked.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NFTMediaInfoStacked.vue:71]:
```suggestion
const isMinimal = computed(() => props.variant.includes('minimal'))
```

File: libs/ui/src/components/NeoNftCard/NeoNftCard.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NeoNftCard.vue:110]:
```suggestion
const isStacked = computed(() => props.variant.includes('stacked'))
```

File: libs/ui/src/components/NeoNftCard/NeoNftCard.vue
Comment (preschian) [inline libs/ui/src/components/NeoNftCard/NeoNftCard.vue:117]:
```suggestion
const isMinimal = computed(() => props.variant.includes('minimal'))
```

**Review by preschian — APPROVED** [review]
otherwise lgtm 🚀

**Comment by daiagi** [comment]
Thanks @preschian .
All your comments have been implemented
waiting on you to check again and resolve the conversations

**Review by prury — APPROVED** [review]

**Comment by exezbcz** [comment]
image
small detail - small caps the "x" 

text truncation issue on your collection sir (if the collection name is too long it resizes the card) 

image

**Comment by daiagi** [comment]
@exezbcz waiting on your approval sir

**Comment by exezbcz** [comment]
working nice! looking good for me

image

**Comment by exezbcz** [comment]
who merge?

**Comment by daiagi** [comment]
> who merge? 


@yangwao

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.04 USD/DOT ~ 12.376 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x7bbff6dcac3af4ea6f1fbd1f36566f584974b8faa58919980afcd2a216699a94](https://polkadot.subscan.io/extrinsic/0x7bbff6dcac3af4ea6f1fbd1f36566f584974b8faa58919980afcd2a216699a94)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**
